loleaflet/src/core/Socket.js

```diff
@@ -1389,8 +1389,15 @@ app.definitions.Socket = L.Class.extend({
 		this._map.fire('docloaded', {status: false});
 
 		if (!this._reconnecting) {
-			this._reconnecting = true;
-			this._map._activate();
+			// We need to make sure that the message slurping processes the
+			// events first, because there could have been a message like
+			// "close: idle" from the server.  Without this, we'd immediately
+			// reconnect (because the "close: idle" was not processed yet).
+			var that = this;
+			setTimeout(function () {
+				that._reconnecting = true;
+				that._map._activate();
+			}, 1 /* ms */);
```

Comment:
Maybe we can give the server a bit more time, like 50ms or 500ms? Otherwise, looks great, thanks!

Reply:
As we talked - I believe the problem is JS only; if the server needed more time, it would have been a problem even in the pre-message-slurping era - so I hope we are fine here.